internal/cluster/validations/validations.go

```diff
@@ -312,73 +315,298 @@ func ValidateVipDHCPAllocationWithIPv6(vipDhcpAllocation bool, machineNetworkCID
+			targetConfiguration.APIVips = []*models.APIVip{}
 		}
-		if c.VipDhcpAllocation != nil {
-			vipDhcpAllocation = *c.VipDhcpAllocation
+		// APIVip was changed (but not cleared), APIVips will be forcefully set to the value of APIVip as a one-element list.
+		if swag.StringValue(params.APIVip) != "" {
+			targetConfiguration.APIVips = []*models.APIVip{{IP: models.IP(swag.StringValue(params.APIVip)), ClusterID: *cluster.ID}}
 		}
-		targetConfiguration.ClusterNetworks = c.ClusterNetworks
-		targetConfiguration.ServiceNetworks = c.ServiceNetworks
-		targetConfiguration.MachineNetworks = c.MachineNetworks
-	case *models.V2ClusterUpdateParams:
-		if c.APIVip != nil {
-			apiVip = *c.APIVip
+	}
+
+	if cluster.IngressVip != "" {
+		// APIVip was cleared and APIVips are not provided, clear both fields.
+		if swag.StringValue(params.IngressVip) == "" && len(params.IngressVips) == 0 {
+			targetConfiguration.IngressVip = ""
+			targetConfiguration.IngressVips = []*models.IngressVip{}
 		}
-		if c.IngressVip != nil {
-			ingressVip = *c.IngressVip
+		// IngressVip was changed (but not cleared), IngressVips will be forcefully set to the value of APIVip as a one-element list.
+		if swag.StringValue(params.IngressVip) != "" {
+			targetConfiguration.IngressVips = []*models.IngressVip{{IP: models.IP(swag.StringValue(params.IngressVip)), ClusterID: *cluster.ID}}
 		}
-		machineNetworks = c.MachineNetworks
-		if c.UserManagedNetworking != nil {
-			userManagedNetworking = *c.UserManagedNetworking
+	}
+	return validateIPAddresses(ipV6Supported, targetConfiguration)
+}
+
+func VerifyParsableVIPs(apiVips []*models.APIVip, ingressVips []*models.IngressVip) error {
+	var multiErr error
+
+	for i := range apiVips {
+		if string(apiVips[i].IP) != "" && net.ParseIP(string(apiVips[i].IP)) == nil {
+			multiErr = multierror.Append(multiErr, errors.Errorf("Could not parse VIP ip %s", string(apiVips[i].IP)))
 		}
-		if c.VipDhcpAllocation != nil {
-			vipDhcpAllocation = *c.VipDhcpAllocation
+	}
+	for i := range ingressVips {
+		if string(ingressVips[i].IP) != "" && net.ParseIP(string(ingressVips[i].IP)) == nil {
+			multiErr = multierror.Append(multiErr, errors.Errorf("Could not parse VIP ip %s", string(ingressVips[i].IP)))
 		}
-		targetConfiguration.ClusterNetworks = c.ClusterNetworks
-		targetConfiguration.ServiceNetworks = c.ServiceNetworks
-		targetConfiguration.MachineNetworks = c.MachineNetworks
+	}
+	if multiErr != nil && !strings.Contains(multiErr.Error(), "0 errors occurred") {
+		return multiErr
 	}
 
-	allAddresses = append(allAddresses, swag.String(ingressVip), swag.String(apiVip))
-	allAddresses = append(allAddresses, common.GetNetworksCidrs(obj)...)
+	return nil
+}
 
-	err := ValidateIPAddressFamily(ipV6Supported, allAddresses...)
+func validateIPAddressesInput(targetConfiguration common.Cluster) error {
+	if len(targetConfiguration.APIVips) > 2 {
+		return errors.Errorf("api_vips supports 2 vips. got: %d", len(targetConfiguration.APIVips))
```

Comment:
Like somewhere above, in the KubeAPI we use `apiVIPs` and not `api_vips`, so there is a potential for a confusion